rs/src/calls.rs

```diff
@@ -7,12 +7,12 @@ use core::{future::Future, marker::PhantomData};
 pub trait Action {
     type Args;
 
+    fn with_gas_limit(self, gas_limit: GasUnit) -> Self;
```

Comment:
Should it be `Option<GasUnit>` as the `gas_limit` returns an option? Plus otherwise I won't be able to reset it back to initial/default value

Reply:
Leave it as it is